.github/workflows/ibis-rebase-nightly.yml

```diff
@@ -0,0 +1,82 @@
+name: Update Dependencies
+on:
+  schedule:
+    # run every 24 hours at 1 AM
+    - cron: "1 */24 * * *"
+  workflow_dispatch:
+
+jobs:
+  generate_branches:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - name: output branches list
+        id: set-matrix
+        run: |
+          set -euo pipefail
+
+          branches=$(git ls-remote --heads https://github.com/ibis-project/ibis.git \
+              | sed -e 's#\t# #g' -e 's#refs/heads/##g' \
```

Comment:
In the future we should do the rebasing only for newer than actual release versions, so we can keep branches for earlier releases as well. We can handle this later though.